File: apps/mobile-portfolio/src/infrastructure/repositories/AccountRepository.ts
import type { IAccountRepository } from "../../domain/repositories/IAccountRepository";
import type { AccountResource } from "../../domain/entities/accountResource";
import { httpClient } from "@/src/infrastructure/network/httpClient";

export class AccountRepository implements IAccountRepository {
  async list(limit = 100, offset = 0): Promise<AccountResource[]> {
    return httpClient.getList<AccountResource>("accounts", limit, offset);
  }

  async getById(accountId: string): Promise<AccountResource | null> {
    return httpClient.getById<AccountResource>("accounts", accountId);
  }
}

File: apps/server-python/src/core/application/ports/repositories/account_repository.py
from abc import ABC, abstractmethod
from typing import List, Optional
from uuid import UUID

from src.core.domain.entities.account import Account


class IAccountRepository(ABC):
    @abstractmethod
    async def get_by_id(self, account_id: UUID) -> Optional[Account]:
        pass

    @abstractmethod
    async def list(self, limit: int = 100, offset: int = 0) -> List[Account]:
        pass

    @abstractmethod
    async def add(self, account: Account) -> Account:
        pass

    @abstractmethod
    async def save(self, account: Account) -> Optional[Account]:
        pass

    @abstractmethod
    async def remove(self, account_id: UUID) -> bool:
        pass
